org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/internal/ZigBeeHandlerFactory.java

```diff
@@ -71,4 +76,13 @@ protected void setZigBeeChannelConverterFactory(ZigBeeChannelConverterFactory zi
     protected void unsetZigBeeChannelConverterFactory(ZigBeeChannelConverterFactory zigbeeChannelConverterFactory) {
         this.zigbeeChannelConverterFactory = null;
     }
+
+    @Reference(cardinality = ReferenceCardinality.MULTIPLE, policy = ReferencePolicy.DYNAMIC)
+    public void addZigBeeThingTypeProvider(ZigBeeThingTypeProvider zigBeeThingTypeProvider) {
+        thingTypeUIDs.addAll(zigBeeThingTypeProvider.getThingTypeUIDs());
```

Comment:
I dont care - since performance is not actually an issue I am fine to follow your proposal (although I dont believe that we will ever have two providers for the same thing type)